nmap_test.go

```diff
@@ -181,7 +181,7 @@ func TestRun(t *testing.T) {
 				panic(err) // this is never supposed to err, as we are testing run and not new.
 			}
 
-			result, err := s.Run()
+			result, _, err := s.Run()
```

Comment:
We should test that the warnings work as expected though. If you don't know how to do this, let me know and I'll gladly write the unit tests 👍

Reply:
Agreed